### ISSUE TITLE
[MiniDiscussion] RRHF is similar to imitation learning

### ISSUE THREAD

**Issue by mickelliu** [GanjinZero/RRHF#4]
I think RRHF possesses many similarities to the concept of imitation learning:
1. $L_{rank}$: Find the highest ranked policy or "player" and calculate the distance in between. So it is the distance to the "expert"
2. $L_{ft}$: a cross-entropy loss, how likely the model is able to generate the expert's response.

**Comment by Yuanhy1997** [comment]
In my opinion, imitation learning resembles conventional supervised fine-tuning (or SFT) more, since human-generated demonstrations fit the terms of "expert" more than model-generated responses. I think the performance "upper bound" is defined by the "expert" samples while in RRHF the "upper bound" is defined by the performance of reward models. RRHF can help the policy outperform "expert" data. 
I partly agree with the idea that RRHF and imitation learning have similarities. But I won't refer to the top-rank responses as "expert", especially at the early stage of tuning.

**Comment by Yuanhy1997** [comment]
Sry, I kinda forget my manners. We really appreciate the discussion and your comments.

**Comment by Yuanhy1997** [comment]
For the first question, you are right, the reward model including used Dahoas/gptj-rm-static ckpt returns a scalar value for the responses and we rank the response according to the scalar value.   
For the second question, I think your "human feedback" refers to the human-written responses, and it is not always assigned to the max reward score. The reasons, in my opinion, are first the reward model is not a golden standard, it may not always distinguish the best one, second, the policy model is capable of generating responses that are better than human annotations. Also, in case there is a misunderstanding, our "human feedback" refers to the reward model which is a proxy human feedback, since including real-humans is really expensive.

**Comment by mickelliu** [comment]
Thanks for your information.
